src/v/cloud_storage/remote.h

```diff
@@ -133,6 +133,12 @@ class remote {
       const std::optional<size_t>& max_keys,
       retry_chain_node& parent);
 
+    /// Checks if the segment exists in the bucket
+    ss::future<download_result> segment_exists(
```

Comment:
can we also add a test? looks like there also isn't any callers of this method in the PR